### PR TITLE
Add MKE and CVE-2024-21626: Understanding the Potential Impact and Mitigation

### PR DESCRIPTION
**Pull request by ry4nz**
Add 0011 Advisories

### CODE REVIEW

File: advisories/0011.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+### Mirantis Kubernetes Engine (MKE) and CVE-2024-21626: Understanding the Impact and Mitigation
+#### Release Date
+2024-02-08
+#### Overview
+This advisory addresses the impact of CVE-2024-21626, a critical vulnerability identified within runc, and its implications for the Mirantis Kubernetes Engine (MKE). Given that MKE relies heavily on the Mirantis Container Runtime (MCR), this vulnerability directly impacts its operational integrity. We aim to clarify the situation, explain the mitigation steps, and provide actionable guidance for MKE users.
```
Comment (corhere) [inline advisories/0011.md]:
```suggestion
This advisory addresses the impact of CVE-2024-21626, a vulnerability identified within runc, and its implications for the Mirantis Kubernetes Engine (MKE). Given that MKE relies heavily on the Mirantis Container Runtime (MCR), this vulnerability directly impacts its operational integrity. We aim to clarify the situation, explain the mitigation steps, and provide actionable guidance for MKE users.
```
It's not a critical vulnerability; see line 11 of this very document.

File: advisories/0011.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  
+   While CVE-2024-21626 might appear in vulnerability scans of your MKE images due to a dependent library, an update to MKE images is not currently required. MKE's architecture mitigates the vulnerability, and there's no known exploit path within MKE. We understand this discrepancy can be confusing. To address this and enhance transparency, we're actively working to remove unnecessary references to the vulnerability from future MKE image scans. This update is planned for upcoming MKE releases.
+
+#### How to check if MKE is vulnerable
+1. [Download and config the client bundle](https://docs.mirantis.com/mke/3.7/ops/access-cluster/download-configure-client-bundle.html)
+2. Check the MCR version by
+   ```
+   docker version --format "{{range .Server.Components}}{{if eq .Name \"Engine\"}}{{.Version}}{{end}}{{end}}"
+   ```
+3. The MKE cluster is vunerable if and only if the version <= 23.0.9.
```
Comment (corhere) [inline advisories/0011.md]: Check the runc version, not Engine.

File: advisories/0011.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+### Vulnerability Information
+#### CVE Identifier
+CVE-2024-21626
+#### CVSSv3.1
+8.6 (High) CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H
+#### CWEs
+CWE-403, CWE-668
+
+#### Affected Product
+* MKE deployment requires installation of MCR.  Older versions of MCR (<= 23.0.9) contain a vulnerability in the underlying runc component. Please see [the MCR advisory](https://github.com/Mirantis/security/blob/main/advisories/0010.md) for details. Any deployment of MKE with MCR version <= 23.0.9 must upgrade to MCR 23.0.9-1 or later to mitigate this risk.
```
Comment (corhere) [inline advisories/0011.md]:
```suggestion
* MKE deployment requires installation of MCR.  Older versions of the MCR packages contain a vulnerable version of the runc component. Please see [the MCR advisory](https://github.com/Mirantis/security/blob/main/advisories/0010.md) for details.
```

The vulnerable versions and patching steps have some nuance that is neither correctly nor adequately captured here.

File: advisories/0011.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+##### Upgrading manually:
+
+If you prefer a more granular approach, refer to the Mirantis documentation for specific package installation and upgrade instructions on your chosen Linux distribution: [Install MCR on Linux distros - Mirantis Container Runtime](https://docs.mirantis.com/mcr/23.0/install/mcr-linux.html)
+
+##### Important notes:
+
+* Carefully review the documentation before proceeding, as specific steps may vary depending on your cluster configuration and operating system.
+* Thoroughly test your applications after the upgrade to ensure compatibility with the new MCR version.
+* Consider backing up your cluster configuration before making any significant changes.
```
Comment (corhere) [inline advisories/0011.md]:
```suggestion
```

Leave the instructions to the Engine team, please.

File: advisories/0011.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+CWE-403, CWE-668
+
+#### Affected Product
+* MKE deployment requires installation of MCR.  Older versions of MCR (<= 23.0.9) contain a vulnerability in the underlying runc component. Please see [the MCR advisory](https://github.com/Mirantis/security/blob/main/advisories/0010.md) for details. Any deployment of MKE with MCR version <= 23.0.9 must upgrade to MCR 23.0.9-1 or later to mitigate this risk.
+
+* No MKE Image Update Needed for CVE-2024-21626
+  
+   While CVE-2024-21626 might appear in vulnerability scans of your MKE images due to a dependent library, an update to MKE images is not currently required. MKE's architecture mitigates the vulnerability, and there's no known exploit path within MKE. We understand this discrepancy can be confusing. To address this and enhance transparency, we're actively working to remove unnecessary references to the vulnerability from future MKE image scans. This update is planned for upcoming MKE releases.
+
+#### How to check if MKE is vulnerable
```
Comment (corhere) [inline advisories/0011.md:22]: But you just said MKE was not vulnerable...

File: advisories/0011.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+8.6 (High) CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H
+#### CWEs
+CWE-403, CWE-668
+
+#### Affected Product
+* MKE deployment requires installation of MCR.  Older versions of MCR (<= 23.0.9) contain a vulnerability in the underlying runc component. Please see [the MCR advisory](https://github.com/Mirantis/security/blob/main/advisories/0010.md) for details. Any deployment of MKE with MCR version <= 23.0.9 must upgrade to MCR 23.0.9-1 or later to mitigate this risk.
+
+* No MKE Image Update Needed for CVE-2024-21626
+  
+   While CVE-2024-21626 might appear in vulnerability scans of your MKE images due to a dependent library, an update to MKE images is not currently required. MKE's architecture mitigates the vulnerability, and there's no known exploit path within MKE. We understand this discrepancy can be confusing. To address this and enhance transparency, we're actively working to remove unnecessary references to the vulnerability from future MKE image scans. This update is planned for upcoming MKE releases.
```
Comment (corhere) [inline advisories/0011.md]: > MKE's architecture mitigates the vulnerability, and there's no known exploit path within MKE.

That statement could be easily misinterpreted as "MKE protects you from the runc vulnerability." MKE _components_ are not vulnerable to this particular exploit, but MKE _does not protect users from harm_ if they ran a malicious container image with a vulnerable runc installed.

**Review by corhere — CHANGES_REQUESTED** [review]

File: advisories/0011.md (hunk 2 above)
Comment (ry4nz) [reply advisories/0011.md]: While I understand the root cause lies with runc, MKE customers interact directly with MCR, not runc. Therefore, focusing on MCR version checks seems more relevant.

Comment (corhere) [reply advisories/0011.md]: Users will remain vulnerable if they upgrade their docker-ee package without upgrading their containerd.io package, as is possible (at least) on RPM distros. The _only_ way to tell for sure is to check the version of runc, **which docker version already conveniently outputs**

File: advisories/0011.md (hunk 3 above)
Comment (ry4nz) [reply advisories/0011.md]: While you could technically install individual packages like Docker-EE, Docker-EE-CLI, and containerd, Mirantis does not officially support such a configuration. Customers and Mirantis typically and should always treat MCR as a single, integrated product that includes all these components pre-configured and tested for compatibility. Using MCR ensures a supported and well-maintained environment, avoiding potential issues arising from manual package installation and management.

File: advisories/0011.md (hunk 6 above)
Comment (ry4nz) [reply advisories/0011.md]: Let me rewrite this paragraph

File: advisories/0011.md (hunk 2 above)
Comment (ry4nz) [reply advisories/0011.md]: I may add a check to `runc` and perhaps an after-upgrade check as well.  However please check https://github.com/Mirantis/security/pull/4/files#r1480921670 for my understanding of MCR as a product.

File: advisories/0011.md (hunk 5 above)
Comment (ry4nz) [reply advisories/0011.md:22]: MKE as a product, is vulnerable due to it's dependency to MCR.  On the other hand, MKE images are not vulnerable and thus an update is not necessary.

File: advisories/0011.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+#### How to check if MKE is vulnerable
+1. [Download and config the client bundle](https://docs.mirantis.com/mke/3.7/ops/access-cluster/download-configure-client-bundle.html)
+2. Check the MCR version by
+   ```
+   docker version --format "{{range .Server.Components}}{{if eq .Name \"Engine\"}}{{.Version}}{{end}}{{end}}"
+   ```
+3. The MKE cluster is vunerable if and only if the version <= 23.0.9.
```
Comment (byDimasik) [inline advisories/0011.md]:
```suggestion
3. The MKE cluster is vulnerable if and only if the version <= 23.0.9.
```

File: advisories/0011.md (hunk 2 above)
Comment (corhere) [reply advisories/0011.md]: We want users to always treat MCR as a single integrated product but we distribute it as four separate DEB/RPM packages. There are all sorts of ways that users may get their deployments into an unsupported state, intentionally or by accident. Users do weird things, make mistakes, or they might even do everything right but fall victim to a software bug during upgrade. Users are going to use these instructions to to verify that they have applied the update correctly. Instructions which incorrectly tell users they are not vulnerable if they applied the update incorrectly are worse than useless, _they put our customers at risk_.

File: advisories/0011.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+* No MKE Image Update Needed
+  
+   While CVE-2024-21626 might appear in vulnerability scans of your MKE images due to the dependent library, an update to MKE images is not currently required.  We understand this discrepancy can be confusing. To address this and enhance transparency, we are actively working to remove unnecessary references to the vulnerability from future MKE image scans. This update is planned for upcoming MKE releases.
+
+#### How to check if MKE is vulnerable
+1. [Download and config the client bundle](https://docs.mirantis.com/mke/3.7/ops/access-cluster/download-configure-client-bundle.html)
+2. Check the `runc` version by
+   ```
+   docker version --format "{{range .Server.Components}}{{if eq .Name \"runc\"}}{{.Version}}{{end}}{{end}}"
```
Comment (corhere) [inline advisories/0011.md]:
```suggestion
   docker version --format '{{range .Server.Components}}{{if eq .Name "runc"}}{{.Version}}{{end}}{{end}}'
```

File: advisories/0011.md (hunk 5 above)
Comment (corhere) [reply advisories/0011.md:22]:
```suggestion
#### How to check if your MKE cluster is vulnerable
```

File: advisories/0011.md (hunk 8 above)
Comment (corhere) [inline advisories/0011.md]: Wouldn't that only check the runc version installed on one of the manager nodes? Or does MKE do some magic in the Engine API proxy? How would a user check the rest of the nodes in their cluster?

Comment (ry4nz) [reply advisories/0011.md]: `docker node ls --format {{.EngineVersion}}` to check engine version.  There is no viable way other than `ssh` into nodes.

**Review by corhere — APPROVED** [review]

**Review by ameyag — APPROVED** [review]

**Pull request merged**